compute_endpoint/tests/unit/test_endpointmanager_unit.py

```diff
@@ -930,3 +932,98 @@ def test_pipe_size_limit(mocker, successful_exec, conf_size):
     else:
         assert pyexc.value.code < 87
         assert f"{stdin_data_size} bytes" in mock_log.error.call_args[0][0]
+
+
+@pytest.mark.parametrize(
+    "data",
+    [
+        (True, {"heartbeat": 10}),
+        (True, {"heartbeat": 10, "foo": "bar"}),
+        (False, {}),
+        (False, {"foo": "bar"}),
+    ],
+)
+def test_render_config_user_template(fs, data):
+    is_valid, user_opts = data
+
+    ep_dir = pathlib.Path("config_user.yaml")
+    ep_dir.mkdir(parents=True, exist_ok=True)
+    template = ep_dir / "config_user.yaml"
+    template.write_text("heartbeat_period: {{ heartbeat }}")
+
```

Comment:
That's a weird template path.  "Doesn't matter" for the purposes of this test, but ... oof.  `config_user.yaml/config_user.yaml` ...

Same as well in the following tests.

Reply:
Ah, that's just a typo. I've updated with a more sensible path name.